veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,225 @@
+    - Updates the VMI interface status.
+    - Adds the `MigrationRequired` condition to the VMI with its status set to `False` and a timestamp.
+4. On a future iteration of the reconciliation loop, the VMI controller will update the `MigrationRequired` condition status to `True` (Following no change in the interface statuses for more than TBD seconds after the initial timestamp set on the condition).
+5. The Workload update controller will identify the VMI as a candidate for automatic migration and will create a VMIM object for it.
+6. The Migration controller will remove the `MigrationRequired` condition once the migration is complete.
+7. The vNICs will be hot-unplugged following the migration - same as today.
+
+#### In-Place Hotplug / Hotunplug Flow
+
+1. The VM owner adds one or more new secondary vNICs with bridge binding and matching networks, or sets their state to `absent` to / on a running VM.
+2. The VM controller patches the VMI object with the additional relevant vNICs and networks.
+3. The VMI controller:
+
+    - Patches the virt-launcher pod's Multus network attachment annotation (`k8s.v1.cni.cncf.io/networks`).
+    - Updates the VMI interface status.
+    - Adds the `MigrationRequired` condition to the VMI with the `False` with its status set to `False` and a timestamp.
+4. The dynamic-networks-controller will observe the change to the Multus network attachment annotation and will invoke Multus.
+5. On a future iteration of the reconciliation loop, the VMI controller will observe a change in the interface statuses and will remove the `MigrationRequired` condition.
+6. The virt-handler will set up / remove the necessary plumbing.
+7. The virt-launcher will attach / detach the relevant interfaces to / from the domain.
+
+> [!NOTE]
+> In case one or more SR-IOV NICs are hotplugged - the VMI controller will immediately mark the VMI for automatic migration
+> by adding the `MigrationRequired` condition with the `True` status.
+> This is because the dynamic-networks-controller cannot handle hot-plug of SR-IOV devices.
+> 
+> SR-IOV hotunplug is currently unsupported.
+
+> [!NOTE]
+> The time it takes for:
+> 1. The dynamic-networks-controller to observe the change and invoke Multus.
+> 2. Multus to invoke its delegate CNIs.
+> 3. Delegate CNIs to perform their job.
+> 4. Multus to report its status on the pod's annotations.
+> 5. VMI controller to observe and report the interface statuses.
+> 
+> Is unbounded and may be longer than the timeout defined before an automated migration is requested.
+> It may cause the VMI to migrate even if an in-place operation took place.
+
+
+#### Pros
+
+- Improves the current user experience
+- Will work for hotplugging interfaces using bridge and SR-IOV bindings
+- Keeps the integration with the dynamic-networks-controller
+
+#### Cons
+ - Will not work for non-migratable VMs
+ - Performs a migration for every hotplug / hotunplug
+ - Requires an API change
```

Comment:
If it is only about the new condition, I would add here "minimal" and mention it is only about the status condition.
After-all, the addition is only for controllers to monitor, not for operators who just want this hotplug to work.

Reply:
Done.